src/libs/elektra/cmount.c

```diff
@@ -0,0 +1,2919 @@
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Root mountpoint not possible, because user mountpoint already exists!\n");
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+		if (ksLookupByName (alreadyUsedMountpoints, "system:/", KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Root mountpoint not possible, because system mountpoint already exists!\n");
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+	}
+	else if (keyGetNameSize (mountPoint) > 2 && *mpName == '/')
+	{
+		if (ksLookupByName (alreadyUsedMountpoints, mpName, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Cascading mountpoint %s not possible, because cascading mountpoint %s already exists.\n", mpName, mpName);ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+		char * tmpStr = elektraMalloc (keyGetNameSize (mountPoint) + strlen ("system:"));
+
+		strcpy (tmpStr, "dir:");
+		strcat (tmpStr, mpName);
+		if (ksLookupByName (alreadyUsedMountpoints, tmpStr, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Cascading mountpoint %s not possible, because dir mountpoint already exists.\n", mpName);
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+		strcpy (tmpStr, "user:");
+		strcat (tmpStr, mpName);
+		if (ksLookupByName (alreadyUsedMountpoints, tmpStr, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Cascading mountpoint %s not possible, because user mountpoint already exists.\n", mpName);
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+		strcpy (tmpStr, "system:");
+		strcat (tmpStr, mpName);
+		if (ksLookupByName (alreadyUsedMountpoints, tmpStr, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			fprintf (stderr, "Cascading mountpoint %s not possible, because system mountpoint already exists.\n", mpName);
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+
+		elektraFree (tmpStr);
+	}
+	else
+	{
+		if (ksLookupByName (alreadyUsedMountpoints, mpName, KDB_O_NONE))
+		{
+			// TODO: set error key (was exception in c++ code)
+			// TODO: list used names in error message (see /src/libs/tools/src/backend.cpp:213)
+			fprintf (stderr, "Mountpoint %s is one of the already used names!\n", mpName);
+			ksDel (backendInfos);
+			ksDel (alreadyUsedMountpoints);
+			return false;
+		}
+	}
+
+	ksDel (alreadyUsedMountpoints);
+	ksDel (backendInfos);
+
+	/* TODO: STEP 4: Check if mounted below system:/elektra */
+	Key * elektraCheck = keyDup (mountPoint, KEY_CP_NAME);
+
+	/* Remove namespace */
+	for (int i = 0; i < keyGetNameSize (elektraCheck); ++i)
+	{
+		if (*(mpName + i) == ':')
+		{
+			keySetName (elektraCheck, mpName + i + 1);
+		}
+	}
```

Comment:
Pretty sure the `keyIsBelow*` functions handle cascading keys too. Otherwise, please use:

```suggestion
	/* Remove namespace */
	keySetNamespace (elektraCheck, KEY_NS_CASCADING);
```